### ISSUE TITLE
Encoding of & in youtube URLs makes the downloading with youtube-dl broken

### ISSUE THREAD

**Issue by jopejoe1** [C0D3D3V/Moodle-Downloader-2#103]
## Avoid duplicates
* [x] Bug is not already reported in another issue

## Describe the bug

When downloading a YouTube playlist it download the whole home page of YouTube it try's this URL `https://www.youtube.com/playlist?embed=no&amp;list=PLx2vkxEuABu3turBLmOoUU6YvX8hKiqVd` but it should use this URL `https://www.youtube.com/playlist?embed=no&list=PLx2vkxEuABu3turBLmOoUU6YvX8hKiqVd`

## Steps to reproduce the issue

 
```
Moodle Downloader starting...
Loading config...
Checking for changes for the configured Moodle-Account....
Moodle cookie has expired, an attempt is made to generate a new cookie.
Downloading course information  18/ 18 [Windows 10       |  4665]
Start downloading changed files...
T 0: 100% - Finished!
T 1: 100% - Finished!
T 2: 100% - Finished!
T 3:  82% - https://www.youtube.com/playlist?embed=no&amp;list=PLx2vkxEuABu3turBLmOoUU6YvX8hKiqVd
T 4: 100% - Finished!
Total:  99%    136.51GiB/   136.58GiB | Files:    40/  153 |     ---b/s^C
```

## Technical details
 - OS: Linux (Manjaro)
 - Shell: zsh

## Expected behavior

Downloads the videos in the playlist

## Possible Fix

do not encode `&`

**Comment by C0D3D3V** [comment]
We do not manipulate URLs at any point. The only urls that get maybe manipulated are the URLs, that are in descriptions. Where is this URL located/linked in your moodle couse?

**Comment by jopejoe1** [comment]
Yeah seems like they are located in a description.

![Screenshot from 2021-07-22 21-10-35](https://user-images.githubusercontent.com/34899572/126615738-194d1996-0405-4a44-953b-8ac495f59d6e.png)

**Comment by C0D3D3V** [comment]
I see. Also there is the playlists linked and additional every single video linked. Currently we have no way to prevent downloading the videos twice or multiple times.

We could simple run html decoding on links from descriptions, this should work in most cases, but could break some other links that use special encoded characters

**Comment by C0D3D3V** [comment]
But you can disable playlists at all, in the youtube-dl settings. Its explained in the wiki. Then only the single videos are downloaded.

**Comment by jopejoe1** [comment]
> I see. Also there is the playlists linked and additional every single video linked. Currently we have no way to prevent downloading the videos twice or multiple times.

Thats not a problem as youtube-dl has settings to prevent duplicate downloads 
> 
> We could simple run html decoding on links from descriptions, this should work in most cases, but could break some other links that use special encoded characters

Maybe only decode urls from certain domains?

> But you can disable playlists at all, in the youtube-dl settings. Its explained in the wiki. Then only the single videos are downloaded.

This only works when linking to a video in a playlists not when directly linking to the playlist overview, also have that already enabled.

**Comment by C0D3D3V** [comment]
This is basicly the same issue as #96 and would be fixed with #97.
But this will force redownloading basicly all links that are in descriptions and have an `&` in it.

**Comment by C0D3D3V** [comment]
I implemented #97 I hope this fix your issue. I apologize for the long delay :(

**Comment by C0D3D3V** [comment]
If the error persists, please create a new issue and send the corresponding log file.
